windows.c

```diff
@@ -0,0 +1,128 @@
+#include <stdlib.h>
+#include <windows.h>
+#include <Wininet.h>
+#include <ras.h>
+#include <tchar.h>
+#include <stdio.h>
+#include "common.h"
+
+void reportWindowsError(const char* action) {
+  LPTSTR pErrMsg = NULL;
+  DWORD errCode = GetLastError();
+  FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER|
+      FORMAT_MESSAGE_FROM_SYSTEM|
+      FORMAT_MESSAGE_ARGUMENT_ARRAY,
+      NULL,
+      errCode,
+      LANG_NEUTRAL,
+      pErrMsg,
+      0,
+      NULL);
+  printf("Error %s: %lu %s\n", action, errCode, pErrMsg);
+}
+
+// Figure out which Dial-Up or VPN connection is active; in a normal LAN connection, this should
+// return NULL. NOTE: For some reason this method fails when compiled in Debug mode but works
+// every time in Release mode.
+LPTSTR FindActiveConnection() {
```

Comment:
Hey @oxtoacart I think we should just consolidate everything in winproxy
but doesn't have to happen now.

On Thu, Feb 19, 2015 at 6:37 AM, oxtoacart <notifications@github.com> wrote:

> In windows.c
> <https://github.com/getlantern/pac-cmd/pull/1#discussion_r24991302>:
>
> > +  FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER|
> > +      FORMAT_MESSAGE_FROM_SYSTEM|
> > +      FORMAT_MESSAGE_ARGUMENT_ARRAY,
> > +      NULL,
> > +      errCode,
> > +      LANG_NEUTRAL,
> > +      pErrMsg,
> > +      0,
> > +      NULL);
> > +  printf("Error %s: %lu %s\n", action, errCode, pErrMsg);
> > +}
> > +
> > +// Figure out which Dial-Up or VPN connection is active; in a normal LAN connection, this should
> > +// return NULL. NOTE: For some reason this method fails when compiled in Debug mode but works
> > +// every time in Release mode.
> > +LPTSTR FindActiveConnection() {
>
> @fffw <https://github.com/fffw> @myleshorton
> <https://github.com/myleshorton> Sorry, I'm not 100% clear - what was the
> decision here? If I don't hear otherwise, I'll go ahead and pull the
> pac-cmd changes as-is and open a ticket to consolidate the Windows
> implementations between winproxy and pac-cmd.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/getlantern/pac-cmd/pull/1/files#r24991302>.
>



-- 
--
President
Brave New Software Project, Inc.
https://www.getlantern.org
pgp A998 2B6E EF1C 373E 723F A813 045D A255 901A FD89